src/common/components/select-repository-list/index.js

```diff
@@ -102,37 +102,32 @@ export class SelectRepositoryListComponent extends Component {
     this.props.dispatch(fetchUserRepositories(pageNumber));
   }
 
-  pageSlice(array, pageLinks) {
+  pageSlice(repositoriesIndex, pageLinks) {
```

Comment:
For me the problems with this method lies in the state it has to deal with, pageLinks needs some work I think, but it could absolutely be a helper or static, but I would do it later as part of dealing with pageLinks I think.